src/circuit.rs

```diff
@@ -2,61 +2,707 @@
+    pub(crate) v_new: Option<NoteValue>,
+    pub(crate) psi_new: Option<pallas::Base>,
+    pub(crate) rcm_new: Option<NoteCommitTrapdoor>,
+    pub(crate) rcv: Option<ValueCommitTrapdoor>,
+}
+
+impl UtilitiesInstructions<pallas::Base> for Circuit {
+    type Var = CellValue<pallas::Base>;
+}
 
 impl plonk::Circuit<pallas::Base> for Circuit {
-    type Config = ();
+    type Config = Config;
     type FloorPlanner = SimpleFloorPlanner;
 
     fn without_witnesses(&self) -> Self {
-        Circuit {}
+        Self::default()
     }
 
     fn configure(meta: &mut plonk::ConstraintSystem<pallas::Base>) -> Self::Config {
-        // Placeholder so the proving key is correctly built.
-        meta.instance_column();
+        // Advice columns used in the Orchard circuit.
+        let advices = [
+            meta.advice_column(),
+            meta.advice_column(),
+            meta.advice_column(),
+            meta.advice_column(),
+            meta.advice_column(),
+            meta.advice_column(),
+            meta.advice_column(),
+            meta.advice_column(),
+            meta.advice_column(),
+            meta.advice_column(),
+        ];
+
+        // Constrain v_old - v_new = magnitude * sign
+        // Either v_old = 0, or anchor equals public input
+        // Constrain v_old = 0 or enable_spends = 1.
+        // Constrain v_new = 0 or enable_outputs = 1.
+        let q_orchard = meta.selector();
+        meta.create_gate("Orchard circuit checks", |meta| {
+            let q_orchard = meta.query_selector(q_orchard);
+            let v_old = meta.query_advice(advices[0], Rotation::cur());
+            let v_new = meta.query_advice(advices[1], Rotation::cur());
+            let magnitude = meta.query_advice(advices[2], Rotation::cur());
+            let sign = meta.query_advice(advices[3], Rotation::cur());
+
+            let anchor = meta.query_advice(advices[4], Rotation::cur());
+            let pub_input_anchor = meta.query_advice(advices[5], Rotation::cur());
+
+            let one = Expression::Constant(pallas::Base::one());
+            let not_enable_spends = one.clone() - meta.query_advice(advices[6], Rotation::cur());
+            let not_enable_outputs = one - meta.query_advice(advices[7], Rotation::cur());
+
+            std::array::IntoIter::new([
+                (
+                    "v_old - v_new = magnitude * sign",
+                    v_old.clone() - v_new.clone() - magnitude * sign,
+                ),
+                (
+                    "Either v_old = 0, or anchor equals public input",
+                    v_old.clone() * (anchor - pub_input_anchor),
+                ),
+                ("v_old = 0 or enable_spends = 1", v_old * not_enable_spends),
+                (
+                    "v_new = 0 or enable_outputs = 1",
+                    v_new * not_enable_outputs,
+                ),
+            ])
+            .map(move |(name, poly)| (name, q_orchard.clone() * poly))
```

Comment:
TODO: replace these by uses of Standard PLONK to reduce the number of gates/selectors.

Reply:
We have decided to remove standard PLONK (as noted in the issue).